### PR TITLE
added controllers and worker jobs for post auditEvents endpoint

### PR DESCRIPTION
**Pull request by shriramshankar**
POST v1/auditEvents endpoint  with/without workers enabled.

### CODE REVIEW

**Comment by shriramshankar** [comment]
@iamigo @pallavi2209 This is ready to be reviewed.

File: api/v1/controllers/auditEvents.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const resultObj = { reqStartTime: req.timestamp };
+    const reqStartTime = Date.now();
+    const auditEvents = req.swagger.params.queryBody.value;
+    const body = { status: 'OK' };
+    if (featureToggles.isFeatureEnabled('enableWorkerProcess')) {
+
+      const jobType = require('../../../jobQueue/setup').jobType;
+      const jobWrapper = require('../../../jobQueue/jobWrapper');
+
+      const payLoad = {};
```
Comment (iamigo) [inline api/v1/controllers/auditEvents.js]: payload is a word so no camelcase required

File: api/v1/controllers/auditEvents.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  postAuditEvents(req, res, next) {
+    const resultObj = { reqStartTime: req.timestamp };
+    const reqStartTime = Date.now();
+    const auditEvents = req.swagger.params.queryBody.value;
+    const body = { status: 'OK' };
+    if (featureToggles.isFeatureEnabled('enableWorkerProcess')) {
+
```
Comment (iamigo) [inline api/v1/controllers/auditEvents.js]: no blank line here

File: api/v1/helpers/nouns/auditEvents.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * api/v1/helpers/nouns/auditEvents.js
+ */
+'use strict';
+
+const AuditEvent = require('../../../../db/index').AuditEvent;
+const m = 'auditevent';
+
+module.exports = {
+  apiLinks: {
+    POST: `Update selected attributes of this ${m}`,
```
Comment (iamigo) [inline api/v1/helpers/nouns/auditEvents.js]: POST: `Insert new audit events`

File: db/model/auditevent.js [hunk 4]
```diff
@@ -37,6 +37,7 @@ module.exports = function auditevent(seq, dataTypes) {
     },
     details: {
       type: dataTypes.JSONB,
+      defaultValue: constants.defaultJSONValue,
```
Comment (iamigo) [inline db/model/auditevent.js]: Adding default does not require a migration?

File: jobQueue/setup.js [hunk 5]
```diff
@@ -45,6 +45,7 @@ module.exports = {
     JOB_CLEANUP: 1,
     PERSIST_SAMPLE_STORE: 1,
     SAMPLE_TIMEOUT: 1,
+    BULKCREATE_AUDITEVENTS: 1,
```
Comment (iamigo) [inline jobQueue/setup.js]: name should be `BULK_CREATE_AUDIT_EVENTS`

Also: we want to be able to process more than one of these at a time (imagine there are dozens of collectors sending POST auditevents every few seconds!), so we should set up a concurrency env var for it

File: tests/api/v1/auditEvents/post.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+      done();
+    })
+    .catch(done);
+  });
+
+  after(u.forceDelete);
+  after(tu.forceDeleteUser);
+  const auditEventObj = u.getAuditEventObject();
+
+  it('OK, simple post', (done) => {
```
Comment (iamigo) [inline tests/api/v1/auditEvents/post.js]: add test for if it's not an array?

File: tests/db/model/auditevent/create.js [hunk 7]
```diff
@@ -103,15 +103,19 @@ describe('tests/db/model/auditevent/create', () => {
     .catch(done);
   });
 
-  it('Error, create auditEvent, null details', (done) => {
+  it('Ok, create auditEvent with no details, details must defaul ' +
```
Comment (iamigo) [inline tests/db/model/auditevent/create.js]: `default`

**Comment by shriramshankar** [comment]
@iamigo @pallavi2209 PR updated. Added the migrations too.

**Pull request merged**